pkg/metadata/metadata.go

```diff
@@ -23,6 +23,9 @@ const (
 	DesiredVersionAnnotation = "windowsmachineconfig.openshift.io/desired-version"
 	// RebootAnnotation indicates the node's underlying instance needs to be restarted
 	RebootAnnotation = "windowsmachineconfig.openshift.io/reboot-required"
+	// UnhealthyAnnotation indicates the node's underlying instance is unhealthy i.e. not ready, scheduling disabled or
+	// meet other unhealthy status conditions like OutOfDisk, MemoryPressure, etc.
+	UnhealthyAnnotation = "windowsmachineconfig.openshift.io/unhealthy"
```

Comment:
This information is already captured as part of the NodeStatus which is managed by the kubelet.
We should not be updating this annotation everytime the status changes. That is what is implied by this annotation existing.

This annotation should be strictly for the status of upgrading.

Reply:
I will continue with the review, mentally replacing unhealthy with upgrading

Reply:
In addition, when we we cordon the node causing scheduling Pods on it to be disabled, it does not imply it is unhealthy but rather WMCO is working on it. We do not want to alarm the end user with the unhealthy verbiage.

Reply:
Makes sense. I had mixed feelings about using `unhealthy` for the anno too. I ended up chasing the idea behind `maxUnhealthyCount`. 

Happy to refactor the anno as: `windowsmachineconfig.openshift.io/upgrading` or `windowsmachineconfig.openshift.io/upgrade-in-progress`.

LMK

Reply:
` windowsmachineconfig.openshift.io/upgrading` is preferred for me as well

Reply:
Same here.